### ISSUE TITLE
Replace cleanup options with svgo-loader?

### ISSUE THREAD

**Issue by tarikjn** [MoOx/react-svg-inline#2]
I seems to me that some options in `cleanup` are better addressed by using `svgo-loader`, any take on this?

**Comment by MoOx** [comment]
I already got issues in the past with svgo changing SVG, but you are probably right.

**Comment by artola** [comment]
Some regex changes that can be useful to clean multiline tags, and tags containing whitespace or attributes:

```
title: /<title[^>]*>([\s\S]*?)<\/title>/gi,
desc: /<desc[^>]*>([\s\S]*?)<\/desc>/gi,
comment: /<!--([\s\S]*?)-->/gi,
defs: /<defs[^>]*>([\s\S]*?)<\/defs>/gi,

```

I have a "similar" code to use SVG icons, and removing these tags is useful, eg:
`<title id="title">bla bla bla</title>`
... then I will inject the right title on the fly.

**Comment by tremby** [comment]
I'm confused about why `<defs>` is listed as "useless". Definitions of gradients, patterns, and reusable shapes are often in there, and so many SVGs will break if it is removed. Maybe I'm misunderstanding the purpose of the "cleanup" parts.

For what it's worth, I'm loading files through image-webpack-loader which incorporates svgo already, so is further cleanup really necessary in this module? Do one thing well, as they say.

**Comment by MoOx** [comment]
Sketch exported files usually have shit in this part. I don't produce a lot of svg by myself and (most of the time) use really simple SVGs. As I already said, I got tons of issues with svgo in the past, that why I used this simple technique. A PR to improve the situation might be welcome.
Note that svgo loader won't probably remove things like width/height/fill etc as it's not intuitive in the first place for an "simple" optimizer to do this kind of changes. So we should probably keep this cleanup at the end.
